pyuvdata/uvdata/uvfits.py

```diff
@@ -1158,8 +1159,18 @@ def write_uvfits(
             hdu.header["PSCAL" + str(i + 1) + "  "] = pscal_dict[key]
             hdu.header["PZERO" + str(i + 1) + "  "] = pzero_dict[key]
 
-        # ISO string of first time in self.time_array
-        hdu.header["DATE-OBS"] = Time(self.time_array[0], scale="utc", format="jd").isot
+        # Create an  astropy.time.Time object from first timestamp.
+        # This is used to generate DATE-OBS and RDATE keywords
+        eloc = EarthLocation(
+            self.telescope_location[0],
+            self.telescope_location[1],
+            self.telescope_location[2],
+            unit="m",
+        )
+        obs_date0 = Time(self.time_array[0], format="jd", scale="utc", location=eloc)
```

Comment:
I've modified the code to use `obs_date0` for a few other calcuations too -- the `EarthLocation` is needed for L1360, which computes apparent time. Here's a summary of where it's used:

```
L1173 hdu.header["DATE-OBS"] = obs_date0.strftime("%Y-%m-%d")
L1354 ant_hdu.header["RDATE"] = obs_date0.strftime("%Y-%m-%d")
L1360 ant_hdu.header["GSTIA0"] = obs_date0.sidereal_time("apparent", "tio").deg
L1366 ant_hdu.header["UT1UTC"] = float(obs_date0.delta_ut1_utc)
```

[The previous code](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/4fba712c51d638ed12b6669b157c2926b152b2a0/pyuvdata/uvdata/uvfits.py#L1341) (see below) got GSTIA0 from an `rdate_obj`, however I think the `rdate_obs.sidereal_time` call would fail if `self.rdate` did not have a location set and `self.gst0 = None`:

```python

        if (self.rdate is None) or (self.rdate == ""):
            rdate_obj = Time(np.floor(self.time_array[0]), format="jd", scale="utc")
        else:
            try:
                rdate_obj = Time(self.rdate, scale="utc")
            except ValueError:
                rdate_obj = Time(np.floor(self.time_array[0]), format="jd", scale="utc")

# ... snip ... #

        if self.gst0 is None:
            ant_hdu.header["GSTIA0"] = rdate_obj.sidereal_time("apparent", "tio").deg
        else:
            ant_hdu.header["GSTIA0"] = self.gst0
```

Reply:
So if one were calculating the local sidereal time, you would be correct, but the description of from the UVFITS memo says:

```
GST at midnight. The value of the GSTIA0 keyword shall be the Greenwich sidereal time in degrees at
zero hours on the reference date for the array in the time system specified by the TIMESYS keyword.
```

Greenwich sidereal time does _not_ require the location in order to calculate. Also note that the GST calculated here is for the start of the track, not midnight (i.e., 00 hours) on the reference date.

Reply:
> So that's true for LST, but the keyword above seems to be for GST, which has no such requirement -- it's not clear from the previous instance of the code, but you can make it explicit by calling `rdate_obj.sidereal_time("apparent", "greenwich")`.
> 
> Also note that the GST calculated here is for the start of the track, not midnight (i.e., 00 hours) on the reference date (which of course was what the old code was _incorrectly_ doing by flooring on the JD).

I always have a small existential crisis when working with FITS headers: _where_ am I? What _is_ the time? 🤯